lib/mayaUsd/render/mayaToHydra/renderGlobals.cpp

```diff
@@ -230,225 +396,549 @@ void _GetFromPlug<TfEnum>(const MPlug& plug, TfEnum& out) {
     out = TfEnum(out.GetType(), plug.asInt());
 }
 
+template <typename T>
+bool _SetOptionVar(const MString& attrName, const T& value) {
+    return MGlobal::setOptionVarValue(attrName, value);
+}
+
+bool _SetOptionVar(const MString& attrName, const bool& value) {
+    return _SetOptionVar(attrName, int(value));
+}
+
+bool _SetOptionVar(const MString& attrName, const float& value) {
+    return _SetOptionVar(attrName, double(value));
+}
+
+bool _SetOptionVar(const MString& attrName, const TfToken& value) {
+    return _SetOptionVar(attrName, MString(value.GetText()));
+}
+
+bool _SetOptionVar(const MString& attrName, const std::string& value) {
+    return _SetOptionVar(attrName, MString(value.c_str()));
+}
+
+bool _SetOptionVar(const MString& attrName, const TfEnum& value) {
+    return _SetOptionVar(attrName, TfEnum::GetDisplayName(value));
+}
+
+bool _SetColorOptionVar(const MString& attrName, const float* values, size_t N) {
```

Comment:
I'll see if the template makes this guy easier to read, but generally curious if dropping the GfVec4 variant all-together is better ?
There is one Storm selection setting that does rgba, but the rest of Hydra & Shading for USD seems to be discouraging any use of rgba.

@mattyjams has an opinion on this ?

Reply:
Hmm - yeah, if we can get away with dropping the vec4, that would be best. Hadn't looked to see if that was an option...

Reply:
Not a particularly well informed one I'm afraid, as I'm not that familiar with the mayaToHydra code, and not really in the loop on the Hydra-side work in this area.

But there has been a trend recently to separate out the handling of color and opacity, for example the change here:
https://github.com/PixarAnimationStudios/USD/commit/332db31e5c0c631e404d5f20c17e8e0d6b30a630

That doesn't extend everywhere yet as you pointed out though, since `UsdImagingGLEngine` and `HdxTaskController` at least so far still expect to be given the selection color as a `GfVec4f`.

So yeah, not sure if that helps at all. :/